modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -638,6 +654,38 @@ private CompletableFuture<Void> waitForMetadataCompleteness(long ts) {
         return executorInclinedSchemaSyncService.waitForMetadataCompleteness(HybridTimestamp.hybridTimestamp(ts));
     }
 
+    private CompletableFuture<Boolean> onZoneReplicaCreated(PartitionReplicaEventParameters parameters) {
+        if (!PartitionReplicaLifecycleManager.ENABLED) {
+            return completedFuture(false);
+        }
+
+        List<CompletableFuture<?>> futs = new ArrayList<>();
+
+        readyToProcessTableStarts.thenRun(() -> {
+            Set<TableImpl> zoneTables = zoneTables(parameters.zoneDescriptor().id());
```

Comment:
So, basically compute inside zoneTable prevents races, right?

Reply:
I had ConcurrentHashMap here, but then delete it - because all access to zoneTables covered by locks. But now, after your questions - I understand, that it was stupid idea) covered zones by id, but not the whole map access.

Fixed.